### PR TITLE
First stab at a comment picture upload feature

### PR DESCRIPTION
**Pull request by KlausTrainer**
I tried to reuse the publisher styles from app/assets/stylesheets/publisher.scss, which is difficult, however, because e.g. ID selectors are used instead of classes.

So I basically adopted the styles from app/assets/stylesheets/publisher.scss in app/assets/stylesheets/comments.scss. This creates some code duplication. In order to avoid this, we'd also need to refactor app/assets/stylesheets/publisher.scss, which I didn't want to touch here in order to keep it focused.

### CODE REVIEW

File: app/assets/stylesheets/comments.scss [hunk 1]
```diff
@@ -28,19 +41,124 @@
       border-left: 3px solid $blue;
       padding-left: 3px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 5

File: app/assets/stylesheets/comments.scss [hunk 2]
```diff
@@ -28,19 +41,124 @@
...
     }
+
+    .publisher-textarea-wrapper {
+      background-color: $white;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 4

File: app/assets/stylesheets/comments.scss [hunk 3]
```diff
@@ -28,19 +41,124 @@
...
+
+      .photodropzone {
+        list-style: none;
+
+        .publisher-photo {
+          float: left;
+          height: 80px;
+          line-height: 80px;
+          list-style: none;
+          margin: 8px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 5

File: app/assets/stylesheets/comments.scss [hunk 4]
```diff
@@ -28,19 +41,124 @@
...
+          overflow: hidden;
+          vertical-align: middle;
+          width: 80px;
+
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 5

File: app/assets/stylesheets/comments.scss [hunk 5]
```diff
@@ -28,19 +41,124 @@
...
+          .x {
+            background-color: $white;
+            border-radius: 25px;
+            color: $black;
+            cursor: pointer;
+            display: none;
+            font-size: 50px;
+            font-weight: bold;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 5

File: app/assets/stylesheets/comments.scss [hunk 6]
```diff
@@ -28,19 +41,124 @@
...
+            padding: 0;
+            position: absolute;
+            right: 0;
+            top: 0;
+          }
+        }
+
+        .markdownIndications {
+          color: $medium-gray;
+          line-height: 30px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Nesting should be no greater than 3, but was 4
- Selector should have depth of applicability no greater than 3, but was 5

File: app/assets/stylesheets/comments.scss [hunk 7]
```diff
@@ -28,19 +41,124 @@
       border-left: 3px solid $blue;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 4

File: app/assets/stylesheets/comments.scss [hunk 8]
```diff
@@ -28,19 +41,124 @@
...
       padding-left: 3px;
     }
+
+    .publisher-textarea-wrapper {
+      background-color: $white;
+      border: 1px solid $border-dark-grey;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 5

File: app/assets/stylesheets/comments.scss [hunk 9]
```diff
@@ -28,19 +41,124 @@
...
+        border-top: 1px solid $white;
+        color: $white;
+        font-weight: bold;
+        text-align: center;
+      }
+
+      .photodropzone {
+        list-style: none;
+
+        .publisher-photo {
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 4

File: app/assets/stylesheets/comments.scss [hunk 10]
```diff
@@ -28,19 +41,124 @@
...
+            background-color: $white;
+            border-radius: 25px;
+            color: $black;
+            cursor: pointer;
+            display: none;
+            font-size: 50px;
+            font-weight: bold;
+            height: 50px;
+            left: 15px;
+            line-height: 50px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 6

File: app/assets/stylesheets/comments.scss [hunk 11]
```diff
@@ -28,19 +41,124 @@
...
+            opacity: .85;
+            position: absolute;
+            text-align: center;
+            top: 15px;
+            width: 50px;
+            z-index: 2;
+
+            &::before {
+              content: '\2716';
+              font-family: 'entypo';
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 7

File: app/assets/stylesheets/comments.scss [hunk 12]
```diff
@@ -28,19 +41,124 @@
...
+            max-height: 20px;
+            max-width: 20px;
+            position: absolute;
+            right: -7px;
+            top: -5px;
+            width: 20px;
+            z-index: 1;
+          }
+
+          &:hover .x {
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 7

File: app/assets/stylesheets/comments.scss [hunk 13]
```diff
@@ -28,19 +41,124 @@
...
+            display: inline-block;
+          }
+
+          img {
+            width: 80px;
+          }
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 7

File: app/assets/stylesheets/comments.scss [hunk 14]
```diff
@@ -28,19 +41,124 @@
...
+        border-top: 1px solid $light-grey;
+
+        .file-upload {
+          direction: ltr;
+          overflow: hidden;
+          position: relative;
+
+          .file-upload-input {
+            margin: 0;
+            opacity: 0;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 7

File: app/assets/stylesheets/comments.scss [hunk 15]
```diff
@@ -28,19 +41,124 @@
...
+            padding: 0;
+            position: absolute;
+            right: 0;
+            top: 0;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 7

File: app/assets/stylesheets/comments.scss [hunk 16]
```diff
@@ -28,19 +41,124 @@
...
+          }
+        }
+
+        .markdownIndications {
+          color: $medium-gray;
+          line-height: 30px;
+          margin-bottom: 0;
+          padding-left: 10px;
+        }
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 6

File: app/assets/stylesheets/comments.scss [hunk 17]
```diff
@@ -28,19 +41,124 @@
...
+      }
+    }
   }
   .submit_button {
     margin-top: 10px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: Selector should have depth of applicability no greater than 3, but was 7

File: app/assets/stylesheets/comments.scss [hunk 18]
```diff
@@ -28,19 +41,124 @@
...
     input {
       float: right;
     }
     padding-left: 12px;
-    display: none;
   }
   .comment_box {
-    height: 35px;
+    border: 0;
     resize: none;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/comments.scss]: - Selector should have depth of applicability no greater than 3, but was 6
- Selector `markdownIndications` should be written in lowercase with hyphens

**Comment by mahonrig** [comment]
This would be a great feature to have, and I'd like to tackle whatever needs done to get this merged. 

Anybody have any feedback for this besides fixing the CSS warnings, merge conflicts, and getting tests written before I get started?

**Comment by Flaburgan** [comment]
Nothing special from my side, I guess @svbergerem 's opinion would be nice.

**Comment by svbergerem** [comment]
After a quick look:

* Rebase
* Switch to FineUploader
* Replace old color variables
* Add photos in comments to the [federation gem](https://github.com/diaspora/diaspora_federation) (talk to @SuperTux88)
* Add backend code (models and controllers)
* Reduce code duplication
* Fix Pronto errors
* Write tests

**Comment by denschub** [comment]
Maybe a discussion if we actually want this? To me, having pictures in comments feel like something that can get messy really fast. But if this is just my opinion, I'm more than happy to ignore it.

**Comment by svbergerem** [comment]
I agree that we should discuss this first. If anyone is interested in implementing this, please create a new thread on [Discourse](https://discourse.diasporafoundation.org/) first.

**Comment by cmrd-senya** [comment]
[Here](https://discourse.diasporafoundation.org/t/comment-picture-upload-feature/821) is a topic for that.

**Comment by Flaburgan** [comment]
It looks like we have an agreement on discourse: users should be able to upload pictures from the comment publisher, and then place them with markdown, like here on github. So no gallery on comment.

**Comment by denschub** [comment]
After a team-internal discussion, let's go ahead and close this PR. There is a lot of work left to do, especially since we recently re-engineered the uploader for posts. This PR would need a large refactoring, at which point most of the code would be new anyway.

If someone is interested in working on this, feel free to pick up this PR and base your own work on the work being done here, but for now, we have no intentions of continuing this work ourselves.

**Pull request closed without merging**